src/TransitionSystems/transition_id.rs

```diff
@@ -12,6 +12,109 @@ pub enum TransitionID {
     None,
 }
 
+impl TransitionID {
+    /// Returns a vector of transitionIDs for all components involved in the transition
+    /// For example
+    /// ```
+    /// use crate::reveaal::TransitionSystems::TransitionID;
+    /// let id = TransitionID::Conjunction(
+    ///     Box::new(TransitionID::Simple("a".to_string())),
+    ///     Box::new(TransitionID::Simple("b".to_string())));
+    /// let leaves = id.get_leaves();
+    /// assert_eq!(leaves, vec![vec![TransitionID::Simple("a".to_string())], vec![TransitionID::Simple("b".to_string())]])
+    /// ```
+    /// Leaves will be {{a}, {b}}, as a is from the first component and b is from the second component
+    pub fn get_leaves(&self) -> Vec<Vec<TransitionID>> {
+        let mut result = Vec::new();
+        self.get_leaves_helper(&mut result, 0);
+        result
+    }
+
+    fn get_leaves_helper(
+        &self,
+        current_leaves: &mut Vec<Vec<TransitionID>>,
+        index: usize,
+    ) -> (&TransitionID, usize) {
+        match self {
+            TransitionID::Conjunction(l, r) | TransitionID::Composition(l, r) => {
+                let a = l.get_leaves_helper(current_leaves, index);
+                let b = r.get_leaves_helper(current_leaves, a.1 + 1);
+                (self, b.1)
+            }
+            TransitionID::Quotient(_l, _r) => {
+                let mut curIndex = index;
+                for t in _l {
```

Comment:
`_l` should not have an underscore as it means it is unused

Reply:
fixed til l